docs/desgin-archive/koordlet-runtime-hooks.md

```diff
@@ -0,0 +1,59 @@
+# Design of Runtime Hook Module in Koordlet
+
+## Summary
+Runtime Hooks acts as the back-end server of Runtime Hook Manager. Runtime Hook Manager is a CRI Proxy, which 
+intercepting the CRI request, calling back-end server to inject policies, such as setting resource isolation 
+parameters by pod priorities, applying resource allocation policies. Runtime Hooks provides a framework to maintain 
+different kinds of policies, and provides flexible extension points during the lifecycle of containers.
+
+## Goals
+- Detailed design of runtime hook module.
+- Developing reference for runtime hook plugin.
+
+## Non-Goals
+- Usage manual for Koordlet users.
+
+## Architecture
+![image](../images/runtime-hooks-arch.svg)
+
+### Modules
+![image](../images/runtime-hooks-modules.svg)
+
+#### Hooks
+`Hooks` module provides a framework for plugins to inject resource parameters for pods and containers. For each new 
+implemented plugin, there are two interfaces need to be implemented: 
+1. register to `Rule` module for policy configuration management.
+2. register to `Hooks` framework for `gRPC Server` calling at corresponding injection stage.
+
+#### gRPC Server
+The `gRPC Server` provides the service for runtime injection, which implements the protocol defined by Runtime Manager.
+After registered the service endpoint to Runtime Manager, the gRPC Server module will receive requests of different 
+injection stage from Runtime Manager, such as `PreRunPodSandboxHook` and `PreStartContainerHook`. For each request,
+the gRPC Server will call corresponding plugins to do the injections, fill the protocol parameters 
+and response to Runtime Manager.
+
+#### Rule
+The `Rule` module is responsible for configuration management, parsing the `NodeSLO` changes and update the local cache
+of injection policies. For each policy update, the `Rule` module also provides callback procedure to refresh the
+parameters of pods and containers.  
+
+### Working Mode
```

Comment:
It is difficult to understand the similarities and differences of the working mechanisms of the two working modes. Can you add a diagram to illustrate the difference between the two modes？